runner/sidecar/sink/kafka/kafka.go

```diff
@@ -41,11 +42,46 @@ func New(ctx context.Context, secretInterface corev1.SecretInterface, x dfv1.Kaf
 	if err != nil {
 		return nil, err
 	}
+
+	producer, err := sarama.NewAsyncProducer(x.Brokers, config)
+	if err != nil {
+		return nil, err
+	}
+
+	logger := sharedutil.NewLogger()
+
 	if x.Async {
-		producer, err := sarama.NewAsyncProducer(x.Brokers, config)
-		if err != nil {
-			return nil, err
-		}
+		config.Producer.Return.Successes = true
+		config.Producer.Return.Errors = true
+
+		// track async success and errors
+		var kafkaMessagesProducedSuccess = promauto.NewCounterVec(prometheus.CounterOpts{
+			Subsystem: "input",
+			Name:      "kafka_produced_success",
+			Help:      "Number of messages successfully produced to Kafka",
+		}, []string{"topic", "type"})
+		var kafkaMessagesProducedErr = promauto.NewCounterVec(prometheus.CounterOpts{
+			Subsystem: "input",
+			Name:      "kafka_produce_errors",
```

Comment:
I think this could just be called "errors"

Reply:
how will we differentiate between consume errors v/s produce errors?

Reply:
consumer errors are always from a source, producer always from a sink